hphp/runtime/vm/jit/smashable-instr-arm.h

```diff
@@ -77,6 +77,22 @@ constexpr size_t kSmashJccTargetOff = 12;
 
 ///////////////////////////////////////////////////////////////////////////////
 
+template<typename T>
+  static void patchInstr(TCA inst, T target) {
+  *reinterpret_cast<T*>(inst) = target;
+  auto const end = reinterpret_cast<char*>(inst + sizeof(T));
+  auto const begin = end - sizeof(T);
+  __builtin___clear_cache(begin, end);
+}
+
+using TCA32 = uint32_t;
+
+template <class T>
+static uint32_t makeTCA32(T t) {
```

Comment:
Also, I think you could probably get away with just doing a `safe_cast<uint32_t>` instead of having this function at all.